numpy/core/arrayprint.py

```diff
@@ -1220,16 +1221,16 @@ def format_float_positional(x, precision=None, unique=True,
 
 
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=None):
         if data.size > 0:
-            max_str_len = max(len(str(np.max(data))),
+            self.max_str_len = max(len(str(np.max(data))),
                               len(str(np.min(data))))
         else:
-            max_str_len = 0
-        self.format = '%{}d'.format(max_str_len)
+            self.max_str_len = 0
+        self.sign = sign
 
     def __call__(self, x):
-        return self.format % x
+        return f"{x:{self.sign}{self.max_str_len}d}"
```

Comment:
I think you should keep the old logic and define the `self.format`, then only change it to `self.format.format(x)`.  Maybe `self.format.format(int(x))` is actually better/faster (only if you can test that it is).